core/src/View/Kokkos_ViewAccessPreconditionsCheck.hpp

```diff
@@ -0,0 +1,161 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#ifndef KOKKOS_IMPL_PUBLIC_INCLUDE
+#include <Kokkos_Macros.hpp>
+static_assert(false,
+              "Including non-public Kokkos header files is not allowed.");
+#endif
```

Comment:
Do we need `#include <Kokkos_Macros.hpp>` if we are just `static_assert(false)`?

Reply:
We don't
Please remove it

Reply:
it is included again in line 27 anyways :-)

Reply:
In the end, every header should have `Kokkos_Macros.hpp` IMHO. That's also what we check in https://github.com/kokkos/kokkos/blob/develop/core/unit_test/headers_self_contained/tstHeader.cpp.

Reply:
Done